src/main/java/gregtech/api/mui/synchandler/GhostCircuitSyncHandler.java

```diff
@@ -0,0 +1,94 @@
+
+import gregtech.api.capability.impl.GhostCircuitItemStackHandler;
+import gregtech.api.recipes.ingredients.IntCircuitIngredient;
+
+import net.minecraft.network.PacketBuffer;
+import net.minecraftforge.items.IItemHandler;
+
+import com.cleanroommc.modularui.utils.MouseData;
+import com.cleanroommc.modularui.value.sync.ItemSlotSH;
+import com.cleanroommc.modularui.widgets.slot.ModularSlot;
+
+import java.io.IOException;
+
+public class GhostCircuitSyncHandler extends ItemSlotSH {
+
+    public GhostCircuitSyncHandler(ModularSlot slot) {
+        super(slot);
+    }
+
+    @Override
+    protected void phantomClick(MouseData mouseData) {
+        if (mouseData.mouseButton == 0) {
+            // increment on left-click
+            setCircuitValue(getNextCircuitValue(true));
+        } else if (mouseData.mouseButton == 1 && mouseData.shift) {
+            // clear on shift-right-click
+            setCircuitValue(GhostCircuitItemStackHandler.NO_CONFIG);
+        } else if (mouseData.mouseButton == 1) {
+            // decrement on right-click
+            setCircuitValue(getNextCircuitValue(false));
+        }
+    }
+
+    @Override
+    protected void phantomScroll(MouseData mouseData) {
+        setCircuitValue(getNextCircuitValue(mouseData.mouseButton == 1));
+    }
+
+    private void setCircuitValue(int value) {
+        GhostCircuitItemStackHandler handler = getGhostCircuitHandler();
+        handler.setCircuitValue(value);
+        syncToClient(1, buf -> {
+            buf.writeBoolean(false);
+            buf.writeItemStack(handler.getStackInSlot(0));
+            buf.writeBoolean(false);
+        });
+    }
+
+    @Override
+    public void readOnServer(int id, PacketBuffer buf) throws IOException {
+        if (id == 10) {
+            setCircuitValue(buf.readShort());
+        } else {
+            super.readOnServer(id, buf);
+        }
+    }
+
+    private int getNextCircuitValue(boolean increment) {
+        GhostCircuitItemStackHandler handler = getGhostCircuitHandler();
+        if (increment) {
+            // if at max, loop around to no circuit
+            if (handler.getCircuitValue() == IntCircuitIngredient.CIRCUIT_MAX) {
+                return GhostCircuitItemStackHandler.NO_CONFIG;
+            }
+            // if at no circuit, skip 0 and return 1
+            if (!handler.hasCircuitValue()) {
+                return 1;
+            }
+            // normal case: increment by 1
+            return handler.getCircuitValue() + 1;
+        } else {
+            // if at no circuit, loop around to max
+            if (!handler.hasCircuitValue()) {
+                return IntCircuitIngredient.CIRCUIT_MAX;
+            }
+            // if at 1, skip 0 and return no circuit
+            if (handler.getCircuitValue() == 1) {
+                return GhostCircuitItemStackHandler.NO_CONFIG;
+            }
+            // normal case: decrement by 1
+            return handler.getCircuitValue() - 1;
+        }
+    }
```

Comment:
I feel like this could be simplified if you passed in `mouseButton` instead of a boolean, since `mouseButton can only be -1 for down or 1 for up`. You could probably do something like this:

```java
    private int getNextCircuitValue(int increment) {
        GhostCircuitItemStackHandler handler = getGhostCircuitHandler();
        // if at max, loop around to no circuit
        if (handler.getCircuitValue() == IntCircuitIngredient.CIRCUIT_MAX) {
                return GhostCircuitItemStackHandler.NO_CONFIG;
        // if at 1, skip 0 and return no circuit
        } else if (handler.getCircuitValue() == 1) {
                return GhostCircuitItemStackHandler.NO_CONFIG;
        // if at 1, skip 0 and return no circuit if decrementing, 
        // or if at no circuit, skip 0 and return 1 when incrementing
        } else if (!handler.hasCircuitValue()) {
                return increment == 1 ? 1 : IntCircuitIngredient.CIRCUIT_MAX
        }
        // normal case: increment by "increment" which is either 1 or -1
        return handler.getCircuitValue() + increment;
    }
```